app/src/main/java/com/keylesspalace/tusky/components/report/Report2Activity.kt

```diff
@@ -0,0 +1,135 @@
+package com.keylesspalace.tusky.components.report
+
+import android.content.Context
+import android.content.Intent
+import android.os.Bundle
+import android.text.Spanned
+import android.view.MenuItem
+import androidx.fragment.app.Fragment
+import androidx.lifecycle.Observer
+import androidx.lifecycle.ViewModelProviders
+import com.keylesspalace.tusky.BaseActivity
+import com.keylesspalace.tusky.R
+import com.keylesspalace.tusky.components.report.adapter.ReportPagerAdapter
+import com.keylesspalace.tusky.di.ViewModelFactory
+import com.keylesspalace.tusky.util.HtmlUtils
+import dagger.android.AndroidInjector
+import dagger.android.DispatchingAndroidInjector
+import dagger.android.support.HasSupportFragmentInjector
+import kotlinx.android.synthetic.main.activity_report2.*
+import kotlinx.android.synthetic.main.toolbar_basic.*
+import javax.inject.Inject
+
+class Report2Activity : BaseActivity(), HasSupportFragmentInjector {
+
+    @Inject
+    lateinit var dispatchingFragmentInjector: DispatchingAndroidInjector<Fragment>
+
+    @Inject
+    lateinit var viewModelFactory: ViewModelFactory
+
+    private lateinit var viewModel: ReportViewModel
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        viewModel = ViewModelProviders.of(this, viewModelFactory)[ReportViewModel::class.java]
+        val accountId = intent?.getStringExtra(ACCOUNT_ID)
+        val accountUserName = intent?.getStringExtra(ACCOUNT_USERNAME)
+        if (accountId.isNullOrBlank() || accountUserName.isNullOrBlank()) {
+            finish()
```

Comment:
Throw an `IllegalStateException` here. I should never happen, but when it does it shows up in crash reports.

Reply:
Done